packages/node/src/generators/library/library.ts

```diff
@@ -89,18 +92,32 @@ function normalizeOptions(tree: Tree, options: Schema): NormalizedSchema {
   };
 }
 
+function getCaseAwareFileName(options: {
+  pascalCaseFiles: boolean;
+  fileName: string;
+}) {
+  const normalized = names(options.fileName);
+
+  return options.pascalCaseFiles ? normalized.className : normalized.fileName;
+}
+
 function createFiles(tree: Tree, options: NormalizedSchema) {
   const nameFormats = names(options.fileName);
+
+  const { className, name, propertyName, fileName } = names(options.fileName);
```

Comment:
You can reuse this line instead:
```suggestion
  const { className, name, propertyName } = names(options.fileName);
```